tidy3d/components/data/monitor_data.py

```diff
@@ -827,7 +828,7 @@ def time_reversed_copy(self) -> FieldTimeData:
         return self.copy(update=new_data)
 
 
-class ModeSolverData(ModeSolverDataset, ElectromagneticFieldData):
+class ModeSolverData(ModeSolverDataset, ElectromagneticFieldData, StubData):
```

Comment:
We won't want to import web things into the `tidy3d/components` because we enforce that the components are "pure" (to avoid circular dependencies).

So what I'm thinking is that in `web` we will need to define an (EM-controlled) class that interface between `Stub` and `StubData` and the tidy3d components. See some of my earlier comments in lark, for example

```py
class Tidy3DStub(Tidy3DBaseModel, Stub):
    component: # eg a Simulation, SimulationData, etc.

    def to_file(self, fname):
         self.components.to_file(fname, ....)
```

Then the webapi can use these interfaces. How does that sound to you?

Reply:
modified